pkg/controller/controller.go

```diff
@@ -575,9 +579,32 @@ func (c *RolloutController) updateStatefulSetPods(ctx context.Context, sts *v1.S
 				continue
 			}
 
-			level.Info(c.logger).Log("msg", fmt.Sprintf("terminating pod %s", pod.Name))
-			if err := c.kubeClient.CoreV1().Pods(pod.Namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{}); err != nil {
-				return false, errors.Wrapf(err, "failed to delete pod %s", pod.Name)
+			if c.zpdbController != nil && c.zpdbController.Running() {
```

Comment:
When would `zpdbController` ever be `nil`, or not be running when we get here?

And if it's not running, should we stop? Otherwise we might stop a pod that would otherwise not be allowed.